### PR TITLE
Refactor: HL API and Package Name

### PR DESCRIPTION
**Pull request by philipaconrad**
## What changed?

High-Level API (`OpaClient`):
 - `input` and `path` swapped ordering.
 - Renamed `query<T>` -> `evaluate<T>`.
 - Added no-input variants for `check` and `evaluate`.
 - Changed `check` to always return `bool` instead of `bool?`.
 - `check` and `evaluate` are `async` now.

Packaging:
 - We now package the entire project under the `Styra.Opa` namespace, with `Styra.Opa` as the high-level API, and `Styra.Opa.OpenApi` as the low-level API.
 
Tests:
 - Updated to match the new package naming scheme + `async` method calling convention.
 
CI:
 - Pull Request action updated to account for new file names.

### CODE REVIEW

File: Styra/Opa/OpaClient.cs [hunk 1]
```diff
@@ -49,118 +49,138 @@ public OpaClient(string serverUrl)
...
     /// </summary>
-    /// <param name="input">The input Dictionary OPA will use for evaluating the rule.</param>
     /// <param name="path">The rule to evaluate. (Example: "app/rbac")</param>
     /// <returns>Result, as a boolean, or nil in the case of a query failure.</returns>
-    /// <remarks>The closest idiomatic type mapping to a JSON Object type for .NET is a Dictionary, so we use that here.</remarks>
-    public bool? check(Dictionary<string, object> input, string path)
+    public async Task<bool> check(string path)
     {
-        return query<bool>(input, path);
+        return await evaluate<bool>(path);
```
Comment (srenatus) [inline Styra/Opa/OpaClient.cs:56]: :+1:

File: Styra/Opa/OpaClient.cs [hunk 2]
```diff
@@ -49,118 +49,138 @@ public OpaClient(string serverUrl)
...
     /// </summary>
-    /// <param name="input">The input string OPA will use for evaluating the rule.</param>
     /// <param name="path">The rule to evaluate. (Example: "app/rbac")</param>
-    /// <returns>Result, as a boolean, or nil in the case of a query failure.</returns>
-    public T? query<T>(string input, string path)
+    /// <returns>Result, as an instance of T, or null in the case of a query failure.</returns>
+    public async Task<T?> evaluate<T>(string path)
     {
-        return queryMachinery<T>(Input.CreateStr(input), path);
+        return await queryMachinery<T>(path, Input.CreateNull());
```
Comment (srenatus) [inline Styra/Opa/OpaClient.cs:123]: 💭 Is this Null a JSON `null` or the C#-equivalent of Go's `nil`?

**Review by srenatus — APPROVED** [review]
I know little about all this, but it looks OK from what I can tell 🙃

File: Styra/Opa/OpenApi/Hooks/HookRegistration.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    /// <summary>
+    /// This file is only ever generated once on the first generation and then is free to be modified.
+    /// Any hooks you wish to add should be registered in the InitHooks function. Feel free to define them
+    /// in this file or in separate files in the Hooks folder.
+    /// </summary>
+    public static class HookRegistration
+    {
+        /// <summary>
+        /// Add hooks by calling `Styra.Opa.OpenApi.Hooks.Register<HookInterface>(myHook);`
+        /// where `I<HookInterface>` is one of the following interfaces defined in HookTypes.cs:
```
[the remark this answers is not in the document]
Comment (philipaconrad) [reply Styra/Opa/OpenApi/Hooks/HookRegistration.cs:13]: I'm 90% sure it's generated, but will do some investigation to be certain about it before merging.

[the remark this answers is not in the document]
Comment (philipaconrad) [reply Styra/Opa/OpenApi/Hooks/HookRegistration.cs:13]: It's automatically generated. :facepalm:

File: Styra/Opa/OpaClient.cs (hunk 2 above)
Comment (philipaconrad) [reply Styra/Opa/OpaClient.cs:123]: More a JSON `null`, I think? :thinking: 

Both the `OpenApi.Result` and `OpenApi.Input` types are generated as [Tagged Unions](https://en.wikipedia.org/wiki/Tagged_union).

The `Input.CreateNull()` constructor basically leaves all of the `Input` object's value fields null, while setting the tag field to explicitly indicate that JSON-level null value is being passed through.

**Pull request merged**